pkg/apis/policy/v1alpha1/clusterimagepolicy_validation.go

```diff
@@ -218,6 +223,36 @@ func (source *Source) Validate(ctx context.Context) *apis.FieldError {
 	var errs *apis.FieldError
 	if source.OCI == "" {
 		errs = errs.Also(apis.ErrMissingField("oci"))
+	} else {
+		var registry string
```

Comment:
Instead of having this duplicated in both validation functions, could we hoist this into a function somewhere else? Also, side note, I was surprised from a quick googling that there's not something that we could use for this.

Reply:
Maybe: https://github.com/google/go-containerregistry/blob/02f47e132cf562183cfd7cf272c0605e6183ba53/pkg/name/repository.go#L71-L72

Reply:
There we go, I poked there but couldn't find it, but looks like that may do it?

Reply:
Yes, I reused the validations from the repository and registry names validations (which are private func). I assume we can accept a registry name and/or a repo name.

Reply:
Do you have any contrary thoughts about it? Otherwise I can move on with this function and move it to a generic place. Or do you want me to adapt the functions in go-containerregistry to fit our needs?

Reply:
What I was thinking was just calling:
```go
_, err := name.NewRepository(foo)
if err != nil {
   return &apis.FieldError{
      Message: err.Error(),
   }
}
```

This is what I've done in the past to access this validation logic.

Reply:
I don't like the function NewRepository() because it only focuses about a repository name, but I would say we need to validate a registry name (if that is what they only provided) e.g. `us.gcr.io`. 


I opened this PR:  https://github.com/google/go-containerregistry/pull/1474

Reply:
I closed this PR  https://github.com/google/go-containerregistry/pull/1474, so this can get fixed asap. I adjusted the code to accept both registry/repo and registry.